llvm/lib/Target/BPF/BPFInstrInfo.td

```diff
@@ -281,6 +302,10 @@ defm JSLT : J<BPF_JSLT, "s<", BPF_CC_LT, BPF_CC_LT_32>;
 defm JSLE : J<BPF_JSLE, "s<=", BPF_CC_LE, BPF_CC_LE_32>;
 defm JSET : J<BPF_JSET, "&", NoCond, NoCond>;
 def JCOND : JMP_JCOND<BPF_JCOND, "may_goto", []>;
+
+let isIndirectBranch = 1 in {
+  def JX : JMP_IND<BPF_JA, "gotox", [(brind i64:$dst)]>;
+}
```

Comment:
nice to see how it should be done, I just had hardcoded it in my test branch: https://github.com/aspsk/llvm-project/commit/98773c6a1ade43733c07627fc3b249c94e673a12